src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/DomainEventService.kt

```diff
@@ -207,35 +213,130 @@ class DomainEventService(
       ),
     )
 
-    if (emitDomainEventsEnabled) {
-      val snsEvent = SnsEvent(
-        eventType = typeName,
-        version = 1,
-        description = typeDescription,
-        detailUrl = detailUrl,
-        occurredAt = domainEvent.occurredAt.atOffset(ZoneOffset.UTC),
-        additionalInformation = SnsEventAdditionalInformation(
-          applicationId = domainEvent.applicationId,
+    if (!emitDomainEventsEnabled) {
+      log.info("Not emitting SNS event for domain event because domain-events.cas1.emit-enabled is not enabled")
+      return
+    }
+
+    if (asyncSaveEnabled) {
+      doAsyncSave(
+        domainEvent,
+        typeName,
+        typeDescription,
+        detailUrl,
+        crn,
+        nomsNumber,
+      )
+    } else {
+      doSyncSave(
+        domainEvent,
+        typeName,
+        typeDescription,
+        detailUrl,
+        crn,
+        nomsNumber,
+      )
+    }
+  }
+
+  final fun retryTemplate(): RetryTemplate {
```

Comment:
I think it'd be better to have this at the bottom of the class to save misdirection